examples/sphere/shallow_water.jl

```diff
@@ -256,6 +281,69 @@ elseif test_name == steady_state_compact_test_name
             local_geometry,
         )
 
+        return (h = h, u = u)
+    end
+elseif test_name == barotropic_instability_test_name
+    y0 = map(Fields.local_geometry_field(space)) do local_geometry
+        coord = local_geometry.coordinates
+
+        ϕ = coord.lat
+        λ = coord.long
+
+        if α == 0.0
+            ϕprime = ϕ
+            λprime = λ
+        else
+
+            ϕprime = asind(sind(ϕ) * cosd(α) - cosd(ϕ) * cosd(λ) * sind(α))
+            λprime = asind(sind(λ) * cosd(ϕ) / cosd(ϕprime)) # for alpha45, this experiences numerical precision issues. The test case is designed for either alpha0 or alpha30
+
+            # Temporary angle to ensure λprime is in the right quadrant
+            λcond = cosd(α) * cosd(λ) * cosd(ϕ) + sind(α) * sind(ϕ)
+
+            # If λprime is not in the right quadrant, adjust
+            if λcond < 0.0
+                λprime = -λprime - 180.0 # shifted by 180 compared to the paper, because our λ ∈ [-180, 180]
+            end
+            if λprime < -180.0
+                λprime += 360.0
+            end
+        end
+
+        # Set initial state of velocity field
+        uλprime(ϕprime) =
+            (u_max / eₙ) *
+            exp(1.0 / ((ϕprime - ϕ₀) * (ϕprime - ϕ₁))) *
+            (ϕ₀ < ϕprime < ϕ₁)
+        uϕprime = 0.0
+
+        # Set integral needed for height initial state
+        h_int(γ) =
+            abs(γ) < 90.0 ?
+            (2 * Ω * sind(γ) + uλprime(γ) * tand(γ) / R) * uλprime(γ) : 0.0
+
+        # Set initial state for height field
+        h = h0 - (R / g) * (pi / 180.0) * quadgk(h_int, -90.0, ϕprime)[1]
+
+        if λ > 0.0
+            λ -= 360.0
+        end
+        if λ < -360.0 || λ > 0.0
+            @info "Invalid longitude value"
+        end
+
+        # Add height perturbation
+        h += h_hat * cosd(ϕ) * exp(-(λ^2 / αₚ^2) - ((ϕ₂ - ϕ)^2 / βₚ^2))
```

Comment:
Is this from eq. 4 of the paper? I think λ and ϕ are radians in the paper but are degrees here.

Reply:
The easiest fix would be to scale αₚ and βₚ appropriately.

Reply:
Yeah, that works too.

Reply:
Uh! Thanks for the catch @szy21 !!

Reply:
> The easiest fix would be to scale αₚ and βₚ appropriately.

Yes, I had scaled other quantities appropriately. I had just missed these two. Thanks for the catch!

Reply:
I fixed it in [51a87e8](https://github.com/CliMA/ClimaCore.jl/pull/292/commits/51a87e892b1eb49a4eea503de24e1fda81de2d3c). It didn't change the overall behavior though.